### ISSUE TITLE
signAndEncryptTransaction can send unencrypted messages

### ISSUE THREAD

**Issue by mat888** [SaitoTech/saito-lite-rust#2219]
https://github.com/SaitoTech/saito-lite-rust/blob/292118715bfc68a12de4d28fa84c6f60993b36bb/lib/saito/wallet.ts#L914

If each of these conditions fails, the code following the below excerpt will send the original **unencrypted** `tx.msg`.

```typescript
if (recipient == "") {
        if (this.app.keychain.hasSharedSecret(tx.to[0].publicKey)) {
          tx.msg = this.app.keychain.encryptMessage(tx.to[0].publicKey, tx.msg);
        }
      } else {
        if (this.app.keychain.hasSharedSecret(recipient)) {
          tx.msg = this.app.keychain.encryptMessage(recipient, tx.msg);
        }
      }
```

**Comment by f8daniel** [comment]
Yeah, I think we are fine with the fallback to unencrypted messaging. Better than throwing an error and creating cascading ripples of failures

**Comment by mat888** [comment]
Could then remove the branch which throws an error, and it will send an empty message instead. I understand it's a useful generalization how it is but it's worth thinking about the unintended consequences which can spoil user security.
